File: python/math-and-geometry/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        reversed = digits[::-1]
        carry = 1
        index = 0
        while carry > 0:
            if index == len(reversed):
                reversed.append(carry)
                break
            reversed[index], carry = (reversed[index] + carry) % 10, (reversed[index] + carry) // 10
            index += 1

        return reversed[::-1]

File: c++/math-and-geometry/plus-one.cc
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        std::vector<int> answer = digits;
        std::reverse(answer.begin(), answer.end());
        for (int i = 0; i < answer.size(); ++i) {
            if (answer[i] < 9) {
                answer[i]++;
                break;
            } else {
                answer[i] = 0;
            }
        }
        
        if (answer[answer.size() - 1] == 0) {
            answer.push_back(1);
        }
        std::reverse(answer.begin(), answer.end());

        return answer;
    }
};
